x-pack/filebeat/input/netflow/input.go

```
@@ -83,171 +227,72 @@ func (w *logDebugWrapper) Write(p []byte) (n int, err error) {
+	if !n.started {
+		return
 	}
 
-	var customFields []fields.FieldDict
-	for _, yamlPath := range config.CustomDefinitions {
-		f, err := LoadFieldDefinitionsFromFile(yamlPath)
-		if err != nil {
-			return nil, fmt.Errorf("failed parsing custom field definitions from file '%s': %w", yamlPath, err)
-		}
-		customFields = append(customFields, f)
+	if n.udp != nil {
+		n.udp.Stop()
 	}
-	decoder, err := decoder.NewDecoder(decoder.NewConfig().
-		WithProtocols(config.Protocols...).
-		WithExpiration(config.ExpirationTimeout).
-		WithLogOutput(&logDebugWrapper{Logger: logger}).
-		WithCustomFields(customFields...).
-		WithSequenceResetEnabled(config.DetectSequenceReset).
-		WithSharedTemplates(config.ShareTemplates))
-	if err != nil {
-		return nil, fmt.Errorf("error initializing netflow decoder: %w", err)
-	}
-
-	input := &netflowInput{
-		outlet:           out,
-		internalNetworks: config.InternalNetworks,
-		forwarder:        harvester.NewForwarder(out),
-		decoder:          decoder,
-		logger:           logger,
-		queueSize:        config.PacketQueueSize,
-	}
-
-	input.udp = udp.New(&config.Config, input.packetDispatch)
-	return input, nil
-}
-
-func (p *netflowInput) Publish(events []beat.Event) error {
-	for _, evt := range events {
-		p.forwarder.Send(evt)
-	}
-	return nil
-}
 
-// Run starts listening for NetFlow events over the network.
-func (p *netflowInput) Run() {
-	p.mutex.Lock()
-	defer p.mutex.Unlock()
-
-	if !p.started {
-		logger.Info("Starting UDP input")
-
-		if err := p.decoder.Start(); err != nil {
-			logger.Errorw("Failed to start netflow decoder", "error", err)
-			p.outlet.Close()
-			return
-		}
-
-		p.queueC = make(chan packet, p.queueSize)
-		err := p.udp.Start()
-		if err != nil {
-			logger.Errorf("Error running harvester: %v", err)
-			p.outlet.Close()
-			p.decoder.Stop()
-			close(p.queueC)
-			return
+	if n.decoder != nil {
+		if err := n.decoder.Stop(); err != nil {
+			n.logger.Info("Error stopping decoder", "error", err)
```

[Question]
Why not use level error here? What are the consequences of this decoder failing to stop?

---

@belimawr what is your proposal here? except changing the verbosity of the log do you want to return the stop err?

---

Just changing the log level. It looks pretty odd to me looking for errors at info level.